spec.md

```diff
@@ -2887,25 +2911,17 @@ n must be non-negative and less than or equal to the bit width of e.
 The tail operation truncates the n most significant bits from e.
 n must be non-negative and less than or equal to the bit width of e.
 
-# Flows
-
-An expression's flow partially determines the legality of connecting to and from the expression.
-Every expression is classified as either *source*, *sink*, or *duplex*.
-For details on connection rules refer back to [@sec:connects].
-
-The flow of a reference to a declared circuit component depends on the kind of circuit component.
-A reference to an input port, an instance, a memory, and a node, is a source.
-A reference to an output port is a sink. A reference to a wire or register is duplex.
+# Source and Sink Expressions
 
-The flow of a sub-index or sub-access expression is the flow of the vector-typed expression it indexes or accesses.
+An expression can either be used as either a source expression or a sink expression; if an expression is on the left-hand-side of a connect, it is a sink expression.
+All other expressions are source expressions.
 
-The flow of a sub-field expression depends upon the orientation of the field.
-If the field is not flipped, its flow is the same flow as the bundle-typed expression it selects its field from.
-If the field is flipped, then its flow is the reverse of the flow of the bundle-typed expression it selects its field from.
-The reverse of source is sink, and vice-versa.
-The reverse of duplex remains duplex.
+All hardware components can be referenced with a source expression.
```

Comment:
The work that @mmaloney-sf has been doing on the grammar and general allows this to be simplified.

```suggestion
All expressions may be used as source expressions.
```

As written, it's not specified if primops, muxes, and other things are legal as sources. And they are!

Reply:
Thanks!